taniumthreatresponse_connector.py

```diff
@@ -376,21 +376,17 @@ def _get_filename_from_tanium(self, action_result, file_id):
         return RetVal(phantom.APP_SUCCESS, filename)
 
     def _save_temp_file(self, content):
+        vault_tmp_dir = Vault.get_vault_tmp_dir()
+        temp_dir = os.path.join(vault_tmp_dir, str(uuid.uuid4()))
 
-        if hasattr(Vault, 'get_vault_tmp_dir'):
-            temp_dir = Vault.get_vault_tmp_dir()
-        else:
-            temp_dir = '/opt/phantom/vault/tmp'
-
-        temp_dir = '{}/{}'.format(temp_dir, uuid.uuid4())
         os.makedirs(temp_dir)
 
         # We are getting application/zip object from tanium and it has set default password.
         # So, we are extracting zip object and store it into vault
         with ZipFile(BytesIO(content)) as zobj:
             zobj.extractall(path=temp_dir, pwd=b"infected")
 
-        return temp_dir + '/' + zobj.namelist()[0]
+        return os.path.join(temp_dir, str(zobj.namelist()[0]))
```

Comment:
Is the cast to `str` necessary?

Reply:
Yes, for the proper type checking purpose. `zobj.namelist()[0]` is not `str` so `os.path.join` raises warnings that only strings should be passed